### PR TITLE
[xbmc][fix] Fix problematic format strings.

### PR DESCRIPTION
**Pull request by Paxxi**
Fix crash introduced in #13452
## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by fritsch** [comment]
Runtime test was succesful. We need to check somehow as those requirements are not obvious for the guys used to printf capabilities.

**Comment by Paxxi** [comment]
I'm thinking that for release builds we catch the exception and log the unprocessed format string as it's not severe enough to crash for. In debug I think crashing is reasonable to detect things like this as early as possible.

Any printf syntax will still work and any fmt syntax will work http://fmtlib.net/latest/syntax.html . There will be issues when using `{}` and forgetting to escape them, same really as forgetting to escape `%` in printf but that didn't throw exceptions.
The only unsupported mode is mixing printf style and fmt style format specifiers.

Not sure how best to spread the word but adding doxygen to CLog should help a bit hopefully

**Review by FernetMenta — APPROVED** [review]
thanks

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp [hunk 1]
```diff
@@ -205,7 +205,7 @@ bool CDVDDemuxClient::ParsePacket(DemuxPacket* pkt)
     if (stream->m_context->profile != st->profile &&
         stream->m_context->profile != FF_PROFILE_UNKNOWN)
     {
-      CLog::Log(LOGDEBUG, "CDVDDemuxClient::ParsePacket - {%d} profile changed from %d to %d", st->uniqueId, st->profile, stream->m_context->profile);
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp]: btw: those curly brackets were not my idea. I prefer to keep it simple

**Pull request merged**